### ISSUE TITLE
Migrate Website to Docusaurus 2

### ISSUE THREAD

**Issue by teikjun** [facebook/react-native-website#2146]
## Description
Hi there! The Docusaurus team is currently developing Docusaurus 2, which now has feature parity with React Native's current usage of Docusaurus.

Docusaurus 2 brings about many improvements:

- Client-side rendering with prerendering (site renders without JavaScript!)
- More flexible appearance - Customizable layouts, CSS modules, etc
- Embeddable interactive React components within markdown via MDX

@darshkpatel and I will be helping to migrate and be the POC throughout the migration process. 

You can view the test site here: https://reactnative-v2.netlify.app/docs/

We're keeping track of our progress in #2139.

## To do list
- [x] Run migration CLI 
- [x] Remove unused versions 
- remove from /versioned_docs, /versioned_sidebar, versions.json, docusaurus.config.js
- [x] Fix build errors from all versioned_docs and current docs
- close all unclosed tags, remove invalid tags (handled by #2158)
- for any invalid JSX outside codeblocks, enclose them in codeblocks (handled by #2158)
- use MDX compatible syntax: change `class` to `className`, change `style="..."` to `style={{...}}`. 
- [x] Fix other build errors
- Fix url-loader issue by adding it as dependency in root directory ( as a hack )
- Fix errors in blog
- [x] Migrate navbar items
- [x] Migrate footer items 
- [x] Migrate pages for help, showcase, version 
- replace CompLibrary, siteConfig and other deprecated things with appropriate substitutes
- In docusaurus.config.js, set preset-classic's pages.path to be `'pages/en'` 
- [x] Migrate CSS for pages and docs
- Use CSS modules to limit scope of CSS (global CSS can affect theme components)
- Remove incompatible CSS and fix bugs
- [x] Move assets (handled by #2157)
- Move `/docs/assets` into `/static/docs/assets` 
- Move missing images from `/blog/assets` into `/static/blog/assets` 
- [x] Migrate landing page 
- [ ] Fix CSS bugs in landing page 🔨
- [x] Customize Prism setup and SnackPlayer using remarkablePlugins
- [ ] Fix SnackPlayer bug
- [ ] Fix custom CSS components in docs
- [ ] Fix tabbed items in docs
- [ ] Replace quotes with admonitions in docs
- [ ] Add redirects for moved/renamed docs

**Legends:**

🔨 - in progress  
- [x] - complete

Note: We'll add a more detailed log of changes after completing each item

**Comment by gedeagas** [comment]
Awesome !!

**Comment by Simek** [comment]
@teikjun Can we track the progress back in #2139 instead of MLH repository to be more transparent for the community and make it easier to interact with the changes?

**Comment by teikjun** [comment]
@Simek Sure, we'll do that! I've updated the link in the issue

**Comment by darshkpatel** [comment]
Should we support theme switching? or should I force default to the light theme? v2 supports defaulting to user's system preference too

![Screenshot 2020-08-18 at 12 29 36 PM](https://user-images.githubusercontent.com/11258286/90481216-0d4c6d80-e14f-11ea-93a0-dd6ed7206547.png)
![Screenshot 2020-08-18 at 12 29 41 PM](https://user-images.githubusercontent.com/11258286/90481222-0f163100-e14f-11ea-9777-1cc605c5e400.png)

PS: Links look weird due to a CSS bug, we're working on it.

**Comment by Simek** [comment]
@darshkpatel I think that you should focus on the content and functionality (ex. SnackPlayer, content tabs), due to restricted time of MLH project (fine tuning design and typography might be a lengthy process). 

Generally I would like to see the Dark Mode, but I have not discuss about this with @rachelnabors yet.

P.S. I'm experiencing issue with the disappearing content (after a while) on the page linked in the first post. Is it a known issue?

P.S.2 The second (and biggest problem in my eyes) is that currently doc files migration destroyed the GIT changes history (files were removed and than added back instead of moving them within the repository). This results is that all content is being attributed to the one person, example: 
* https://github.com/MLH-Fellowship/react-native-website/blob/docs/migration/v2-website/docs/accessibility.md
vs
* https://github.com/facebook/react-native-website/blob/master/docs/accessibility.md

**Comment by rachelnabors** [comment]
Please implement the theme switcher—we can handle the design tweaks necessary to make it work. But focus on the functionality: inline toggles for examples and notes.

It would be best if we could keep the contribution records with `git mv` as that information is used for attribution purposes on the page. Please retain it.

Thank you!

**Comment by darshkpatel** [comment]
> P.S. I'm experiencing issue with the disappearing content (after a while) on the page linked in the first post. Is it a known issue?
Thanks for pointing it out! Wasn't aware of it since it doesn't occur while running the dev server. Will add to ToDo

**Comment by slorber** [comment]
That looks quite nice already!

Also see the content disappear, that's what I reported you @rachelnabors , this seems to be related to React hydration as it only happens on first page load, not navigation => https://react-native-v2.netlify.app/docs/

I also think having the theme switcher is great and it seems to work quite nicely already.

Looks like the code blocks highlighting does not work, but I suspect this is because we don't integrate properly the embedded Snack player iframe.

About git, you are totally right, and I was pretty sure this would be a problem for many large v1 website like this one. The migration cli we made is a first version, there are still many things that can be improved, but preserving git history is probably the most critical one. https://github.com/facebook/docusaurus/issues/3103
We'll try to figure out how to preserve it

I also think that for now we should try to see how to add the missing features like the Snack Player etc. Once these are done, we'd have a better idea of how hard it will be to complete this migration project.
